service/controller/resource/instance/create_cordon_old_workers.go

```diff
@@ -131,22 +134,32 @@ func sortNodesByTenantVMState(nodes []corev1.Node, instances []compute.VirtualMa
 		nodeMap[n.GetName()] = n
 	}
 
+	myVersion := semver.New(project.Version())
+
 	for _, i := range instances {
 		name := instanceNameFunc(customObject, *i.InstanceID)
 
-		if *i.LatestModelApplied {
-			n, found := nodeMap[name]
+		n, found := nodeMap[name]
+		if !found {
 			// When VMSS is scaling up there might be VM instances that haven't
 			// registered as nodes in k8s yet. Hence not all instances are
 			// found from node list.
-			if found {
-				newNodes = append(newNodes, n)
-			}
+			continue
+		}
+
+		v, exists := n.GetLabels()[label.OperatorVersion]
+		if !exists {
+			// Label does not exist, this normally happens when a new node is coming up but did not finish
+			// its kubernetes bootstrap yet and thus doesn't have all the needed labels.
+			// We'll ignore this node for now and wait for it to bootstrap correctly.
```

Comment:
Please add it to `newNodes` because otherwise this function doesn't work as expected (`len(input nodes) > (len(oldNodes) + len(newNodes))`) and that might cause problems one day when some maintenance changes are done.

Reply:
This is the check we do
```
if len(newNodes) < len(oldNodes) {
```
The only problem I see with the current "ignore" approach is that the upgrade could be stuck if we remove the label in the future for any reason.

Reply:
Sure, it works now, but theoretically there could be some point in time in the future where the client-side logic expects sum of return values to match input length.

Nevertheless, for now this works correctly.